### ISSUE TITLE
Add option to reveal importable type from `reveal_type`

### ISSUE THREAD

**Issue by skilkis** [microsoft/pyright#3573]
Dear @erictraut, 

I am working on a project where I utilize the awesome type-inferencing capability of Pyright to infer the return types of methods and attributes when a user does not provide a type-annotation. The inferred types are then used in the generation of a GraphQL schema. However, I've run into a limitation with this approach due to how the output of `reveal_type` reports the class name instead of the importable path to the type. This behavior makes it hard to programmatically re-build the type-hint from the return of `reveal_type`. Besides my use-case I believe that this feature would also be welcome for typing tests that use Pyright to check type-hints. In this scenario, a test might incorrectly pass based on class-name equivalence when in reality a different type was imported.

To illustrate this issue I've made an example where there are two files `foo_a.py` and `foo_b.py` that both define a `Foo` class, and a third file `reveal_types.py` where calls to `reveal_type` are made.

**foo_a.py:**
```python
class Foo:
    pass
```
**foo_b.py:**
```python
class Foo:
    pass
```
**reveal_types.py:**
```python
from foo_a import Foo as FooA
from foo_b import Foo as FooB


foo_a = FooA()
foo_b = FooB()
reveal_type(foo_a)
reveal_type(foo_b)
```
In this scenario, Pyright outputs the following:
```
Type of "foo_a" is "Foo"
Type of "foo_b" is "Foo"
```

From the output above, it is not possible to understand what the precise type of `foo_a` or `foo_b` is since `Pylance` seems to return the class name from `reveal_type` which is `Foo` for both cases above.

**Describe the solution you'd like**

Would be great to be able to specify an option such as `reportImportableTypes` or `reportPreciseTypes` in the config file such that `reveal_types` returns an importable type. When `reportPreciseTypes=True`, `Foo` would then be reported as `foo_a.Foo` or `foo_b.Foo`.

**Additional context**
For reference, the desired output of `reveal_type` output of Mypy (v0.961) returns the importable type as follows:
![image](https://user-images.githubusercontent.com/29230371/173399849-246d9c24-a625-483c-a298-f91037ba4131.png)

**Comment by erictraut** [comment]
You're unfortunately using `reveal_type` in a manner in which it is not intended. There are many reasons this approach won't work. That's because `reveal_type` doesn't necessarily output text that is syntactically correct for Python. There are numerous types for which there is no concise syntax available, so pyright (and other type checkers) use a pseudo grammar for these types. Examples include functions with overloads, functions with keyword parameters and defaults, TypeVars and ParamSpecs, etc.

The type that mypy outputs indicates the place where the type was declared, not necessarily the place where it should be imported. Those can differ in some libraries because types are re-exported by modules. The presence of type stubs can also result in inconsistencies here.

Even if these were not issues, I'd still be reluctant to add this feature because it is so niche. It's unlikely it would ever be used by anyone other than you. We generally add new configuration options only if we get signal from users that it will have somewhat broad adoption.

You mentioned this feature might be useful for typing tests. The function `typing.assert_type` was recently added for this purpose. That's a better approach than `reveal_type` for typing tests because `reveal_type` outputs text that varies between type checkers, but `assert_type` operates on expressions.

**Comment by skilkis** [comment]
@erictraut thank you for taking the time to analyze this issue and give an informative response, I do agree that this approach will not work for all cases. Also, its very useful to know that `typing.assert_type` was recently added, it will for sure come in handy!
